ldms/src/sampler/sampler_base.c

```diff
@@ -331,9 +331,17 @@ ldms_set_t base_set_new(base_data_t base)
 {
 	int rc;
 	base->missing_warned = 0;
+	errno = 0;
 	base->set = ldms_set_new(base->instance_name, base->schema);
-	if (!base->set)
+	if (!base->set) {
+		size_t sz = 128;
+		char s[sz];
```

Comment:
I'm not sure how this works at all. Has this been compile tested?

Reply:
this is c99 dynamic array (alloca wrapper syntax). It's equivalent to the
#define X 128
s[X]

It ran (and was needed) as part of dev/test of Omar's lustre sampler update during the stria DST.